src/analysis/type_utils.ml

```diff
@@ -30,6 +30,10 @@ open Std
 
 module Verbosity = Mconfig.Verbosity
 
+let protect expr =
```

Comment:
What does protect do?

Reply:
To get the "generic" type of the operator Merlin needs to parse it and type it. But `let++` is not valid syntax so adding parenthesis `(let++)` is required for it to parse correctly. The protect function uses a function from compiler libs to determine is the expression requires parenthesizing before parsing.